File: SeiSaude-app/src/models/alimento.ts
import { Quantidade } from "./quantidade";
import { Valor } from "./valor";

export class Alimento {
    nome: string;
    complemento: string;
    descricao: string;
    fabricante: string;
    cadastradoPor: string;
    componentes: Quantidade[];
    valNutricionais: Valor[];

    public constructor(nome, complemento, descricao, fabricante, cadastradoPor, componentes, valNutricionais) {
        this.nome = nome;
        this.complemento = complemento;
        this.descricao = descricao;
        this.fabricante = fabricante;
        this.cadastradoPor = cadastradoPor;
        this.componentes = componentes;
        this.valNutricionais = valNutricionais;
    }
}

File: src/main/java/server/entities/Alimento.java
package server.entities;

import javax.persistence.Entity;

@Entity
public class Alimento extends Produto {

	private static final long serialVersionUID = 1L;

	public Alimento(String name, String descricao, String cadastradoPor, String complemento) {
		super(name, descricao, cadastradoPor, cadastradoPor, complemento);
	}

	public Alimento(Elemento elemento, String fabricante) {
		super(elemento, fabricante);
	}
	
	public Alimento() {
	}

}
